src/redux/books/books.js

```diff
@@ -0,0 +1,34 @@
+const ADDED_BOOK = 'mv-bookstore/books/ADDED_BOOK';
+const REMOVED_BOOK = 'mv-bookstore/books/REMOVED_BOOK';
+
+const initialState = [];
+
+// Action Creators
+export const addBook = (title, author) => ({
+  type: ADDED_BOOK,
+  payload: { title, author },
+});
+
+export const removeBook = (id) => ({
+  type: REMOVED_BOOK,
+  payload: id,
+});
+
+// Reducer
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADDED_BOOK:
+      return [
+        ...state,
+        action.payload,
+      ];
+    case REMOVED_BOOK:
+      return [
+        ...state.filter((book) => (book.id !== action.id)),
```

Comment:
- [Optional] A book has the properties title, author, and a unique key(id). We are actually returning an array of new books excluding(filtering) the book(book.id) you want to remove from the store. I don't understand what `action.id` will return. But I'm very sure that `(book.id  !==  id )` will return an array of new books excluding the book you just click on it to remove. If you have any explanation for using  `... !== action.id)`, please let me know else you should go with `(book.id  !==  id )` .


  Thanks :pray:

Reply:
@FrederickMih, thanks for the review 🙏🏽 

In essence, `book.id !== action.id` should return the same array as `book.id !== id`. `action.id` is the _payload_ from the redux action object. Its usage here is similar to the `ADDED_BOOK` _case_ in the `switch` statement, where an array is returned with a new book(`action.payload` / `{title, author}`) appended to the end. 

In hindsight and for consistency sake, `action.id` should have been `action.payload`, which means the return statement should probably be `[ ...state.filter((book) => (book.id !== action.payload)) ]`. 

I've made the necessary change and hope any ambiguity has been cleared.